File: src/main/java/com/example/Toros/request/TournamentSetupRequest.java
package com.example.Toros.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import java.time.LocalDateTime;

@Data
public class TournamentSetupRequest {
    // Basic tournament parameters
    private String tournamentName;
    private int numCourts;
    private int gamesPerTeam;
    private boolean skillBased;
    private LocalDateTime startTime;
    private int matchDuration;
    private int breakTime;
    private Boolean confirmDelete;
    @JsonProperty("auth0UserId")
    private String auth0Id; // Auth0 ID of the organizer

    // Extended tournament properties
    private String location;
    private String organizer;
    private String contactInfo;
    private String tournamentType;
    private String scoringSystem;
    private String rules;
    private String prizeDistribution;
    private String format;
    private String ageGroup;
    private String skillLevel;
}

File: system/backend/API/DTOs/TournamentSetupRequest.cs
namespace Toros.Backend.DTOs;

public class TournamentSetupRequest
{
    public string TournamentName { get; set; } = string.Empty;
    public int NumCourts { get; set; }
    public int GamesPerTeam { get; set; }
    public bool IsSkillBased { get; set; }
    public int MatchDuration { get; set; }
    public int BreakTime { get; set; }
    public DateTime StartTime { get; set; }
    public string Location { get; set; } = string.Empty;
    public string Organizer { get; set; } = string.Empty;
    public string ContactInfo { get; set; } = string.Empty;
    public string TournamentType { get; set; } = "Round Robin";
    public string ScoringSystem { get; set; } = "Rally";
    public string Rules { get; set; } = string.Empty;
    public string PrizeDistribution { get; set; } = string.Empty;
    public string Format { get; set; } = "Doubles";
    public string AgeGroup { get; set; } = string.Empty;
    public string SkillLevel { get; set; } = string.Empty;
    public string? Auth0Id { get; set; }
    public string? UserId { get; set; } // User ID from frontend
    public bool? ConfirmDelete { get; set; }
}
